google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java

```diff
@@ -70,46 +71,82 @@ public CloudSdkPanel() {
         .addDocumentListener(new DocumentAdapter() {
           @Override
           protected void textChanged(DocumentEvent event) {
-            checkSdk();
+            checkSdkInBackground();
           }
         });
 
-    checkSdk();
+    checkSdkInBackground();
   }
 
-  private void checkSdk() {
-    String path = cloudSdkDirectoryField.getText();
+  private void checkSdkInBackground() {
+    final String path = cloudSdkDirectoryField.getText();
+    ApplicationManager.getApplication().executeOnPooledThread(new CloudSdkChecker(path));
```

Comment:
I'd prefer to rename. The main reason I created it was to make it more obvious that it is operating on the cloudSdkPath that is in the text box at the time that this runnable is submitted to the background thread executor.